src/triples/generation.rs

```diff
@@ -441,6 +445,578 @@ async fn do_generation<C: CSCurve>(
     ))
 }
 
+async fn do_generation_many<C: CSCurve, const N: usize>(
+    ctx: Context<'_>,
+    participants: ParticipantList,
+    me: Participant,
+    threshold: usize,
+) -> Result<TripleGenerationOutputMany<C>, ProtocolError> {
+    assert!(N > 0);
+    
+    let mut rng = OsRng;
+    let mut chan = ctx.shared_channel();
+    let mut transcript = Transcript::new(LABEL);
+
+    // Spec 1.1
+    transcript.message(b"group", C::NAME);
+    transcript.message(b"participants", &encode(&participants));
+    // To allow interop between platforms where usize is different
+    transcript.message(
+        b"threshold",
+        &u64::try_from(threshold).unwrap().to_be_bytes(),
+    );
+    
+    let mut my_commitments = vec![];
+    let mut my_randomizers = vec![];
+    let mut ev = vec![];
+    let mut fv = vec![];
+    let mut lv = vec![];
+    let mut big_e_iv = vec![];
+    let mut big_f_iv = vec![];
+    let mut big_l_iv = vec![];
+
+    for _ in 0..N {
+        // Spec 1.2
+        let e: Polynomial<C> = Polynomial::random(&mut rng, threshold);
+        let f: Polynomial<C> = Polynomial::random(&mut rng, threshold);
+        let mut l: Polynomial<C> = Polynomial::random(&mut rng, threshold);
+    
+        // Spec 1.3
+        l.set_zero(C::Scalar::ZERO);
+    
+        // Spec 1.4
+        let big_e_i = e.commit();
+        let big_f_i = f.commit();
+        let big_l_i = l.commit();
+    
+        // Spec 1.5
+        let (my_commitment, my_randomizer) = commit(&mut rng, &(&big_e_i, &big_f_i, &big_l_i));
+        
+        my_commitments.push(my_commitment);
+        my_randomizers.push(my_randomizer);
+        ev.push(e);
+        fv.push(f);
+        lv.push(l);
+        big_e_iv.push(big_e_i);
+        big_f_iv.push(big_f_i);
+        big_l_iv.push(big_l_i);
+    }
+
+    // Spec 1.6
+    let wait0 = chan.next_waitpoint();
+    chan.send_many(wait0, &my_commitments).await;
+
+    // Spec 2.1
+    let mut all_commitments_vec: Vec<ParticipantMap<Commitment>> = vec![];
+    for i in 0..N {
+        let mut m = ParticipantMap::new(&participants);
+        m.put(me, my_commitments[i]);
+        all_commitments_vec.push(m);
+    }
+    
+    while all_commitments_vec.iter().any(|all_commitments| !all_commitments.full()) {
+        let (from, commitments): (_, Vec<_>) = chan.recv(wait0).await?;
+        for i in 0..N {
+            all_commitments_vec[i].put(from, commitments[i]);
+        }
+    }
+    
+    // Spec 2.2
+    let mut my_confirmations = vec![];
+    for i in 0..N {
+        let all_commitments = &all_commitments_vec[i];
+        let my_confirmation = hash(all_commitments);
+        my_confirmations.push(my_confirmation);
+    }
+    
+    // Spec 2.3
+    transcript.message(b"confirmation", &encode(&my_confirmations));
+
+    // Spec 2.4
+    let fut = {
+        let ctx = ctx.clone();
+        let ev0: Vec<_> = ev.iter().map(|e| e.evaluate_zero()).collect();
```

Comment:
I think for consistency changing this to e0_v and (below) f0_v makes sense

Reply:
makes sense will do